### PR TITLE
Update branch manager handbook

### PR DESCRIPTION
**Pull request by hoegaarden**
- Some reformatting and typo fixes
- Add considerations on when to run `branchff`
- Add notes about 1st beta / branch generation
  - staging takes longer
  - test-infra needs to be notified to setup test jobs

/area release-team
/assign @tpepper @spiffxp 
/cc @dougm

xref: #506
Fixes: #509

### CODE REVIEW

File: release-team/role-handbooks/branch-manager/README.md [hunk 1]
```diff
@@ -142,7 +144,23 @@ Run simply as:
 
 This is done daily.
 
-The first time the ‘[branchff](https://git.k8s.io/release/branchff)’ tool is run on a branch will do a clone of k/k to a temporary directory, run a git merge-base, run a few cleanup scripts to insure the branch’s generated API bits are in order (master branch content will move on toward version N+1, while release branch needs to stay at version N), commit the results of those scripts to the branch, and push to the GitHub remote lease branch.  Except again, we have a safety net and the default is a mock run.
+Earlier in the release the exact time of running [`branchff`] might just be at the discretion of the branch manager, as agreed upon with the release lead.
+
+Later in the release it will probably become more important to algin with the release lead and the CI signal team (and probably other release team members).
```
Comment (dougm) [inline release-team/role-handbooks/branch-manager/README.md]: typo: algin -> align

File: release-team/role-handbooks/branch-manager/README.md [hunk 2]
```diff
@@ -142,7 +144,23 @@ Run simply as:
...
+The exact time for pulling in the changes from master to the release branch might depend on the features that have or are to be merged. Considerations could be:
+- We should run [`branchff`] sooner, before `$bigFeature` so we have a signal in the release branch before that fetature was brought in
+- We should run [`branchff`] later, after `$theOtherFeature` has been merged, so we get signal on that feature from both the master and the release branch
+
+
+The first time the [`branchff`] tool is run on a branch it will
+- do a clone of `k/k` to a temporary directory
+- run a git merge-base
+- run a few cleanup scripts to insure the branch’s generated API bits are in order
```
Comment (dougm) [inline release-team/role-handbooks/branch-manager/README.md]: insure -> ensure

File: release-team/role-handbooks/branch-manager/README.md (hunk 1 above)
Comment (hoegaarden) [reply release-team/role-handbooks/branch-manager/README.md]: Thanks -- fixed

File: release-team/role-handbooks/branch-manager/README.md (hunk 2 above)
Comment (hoegaarden) [reply release-team/role-handbooks/branch-manager/README.md]: Thanks -- fixed

**Comment by hoegaarden** [comment]
PTAL, thanks!

**Comment by spiffxp** [comment]
/lgtm
/hold
if you want anyone else to comment, otherwise feel free to /hold cancel

**Comment by hoegaarden** [comment]
thanks!
/hold cancel

**Pull request merged**